### ISSUE TITLE
Unnecessary loop unrolling in twisted/web/_flatten.py

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#5460]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#5460|
|Type|enhancement|
|Created|2012-01-24 14:38:03Z|
|Branch|https://github.com/twisted/twisted/tree/dont-unroll-5460|

The `Tag`-handling case in `_flattenElement` manually unrolls the result of flattening the tag's children.  There's no need to do this manually here, just as there's no need to do it manually anywhere else `_flattenElement` calls itself recursively.

<details><summary>Searchable metadata</summary>

```
trac-id__5460 5460
type__enhancement enhancement
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_dont_unroll_5460 branches/dont-unroll-5460
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__web web
keywords__ 
time__1327415883000000 1327415883000000
changetime__1327686007000000 1327686007000000
version__None None
owner__exarkun exarkun
cc__jknight
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [33484]) Branching to 'dont-unroll-5460'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [33485]) Replace three cases of manual loop unrolling with three simple yield statements.

refs [#5460](https://github.com/twisted/twisted/issues/5460)

**Comment by twisted-trac** [comment]
|[<img alt="MostAwesomeDude's avatar" src="https://avatars.githubusercontent.com/u/118035?s=50" width="50" height="50">](https://github.com/MostAwesomeDude)<a name="note_5"></a>|@MostAwesomeDude set owner to @exarkun|
|-|-|

Looks pretty straightforward. The logic is very recursive and easy to follow. All tests pass, with good code coverage.

This doesn't seem like a feature that needs a big splash of news, so I think it's pretty much ready to go.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [33495]) Merge Twisted dont-unroll-5460

Author: exarkun
Reviewer: MostAwesomeDude
Fixes: [#5460](https://github.com/twisted/twisted/issues/5460)

Slightly simplify the implementation of serializing tags and such to HTML
by removing several occurrences of manual loop unrolling.  These will be
handled by the serialization trampoline automatically.
